src/Workleap.DomainEventPropagation.Subscription.PullDelivery/EventGridClientAdapter/IEventGridClientAdapter.cs

```diff
@@ -0,0 +1,14 @@
+﻿using Azure.Messaging.EventGrid.Namespaces;
+
+namespace Workleap.DomainEventPropagation.EventGridClientAdapter;
+
+internal interface IEventGridClientAdapter
+{
+    Task<IEnumerable<EventGridClientAdapter.EventBundle>> ReceiveCloudEventsAsync(string topicName, string eventSubscriptionName, CancellationToken cancellationToken);
+
+    Task AcknowledgeCloudEventsAsync(string topicName, string eventSubscriptionName, string? lockToken, CancellationToken cancellationToken);
+
+    Task ReleaseCloudEventsAsync(string topicName, string eventSubscriptionName, string? lockToken, CancellationToken cancellationToken);
+
+    Task RejectCloudEventsAsync(string topicName, string eventSubscriptionName, string? lockToken, CancellationToken cancellationToken);
```

Comment:
I think all those methods should use the singular form rather than plural for their naming, given we only act on a single cloud event at a time rather than multiple ones.

```suggestion
    Task AcknowledgeCloudEventAsync(string topicName, string eventSubscriptionName, string? lockToken, CancellationToken cancellationToken);

    Task ReleaseCloudEventAsync(string topicName, string eventSubscriptionName, string? lockToken, CancellationToken cancellationToken);

    Task RejectCloudEventAsync(string topicName, string eventSubscriptionName, string? lockToken, CancellationToken cancellationToken);
```